dbux-projects/src/checkSystem.js

```diff
@@ -80,70 +51,58 @@ function isWindows() {
-    success = false;
-    modalMessage += `x  which/where.exe\n` +
-      `    Cannot run the system check because you are on a legacy system. ` +
-      `We need "where.exe" to be available on Windows (available from Windows 7 and Windows Server 2003 onward), ` +
-      `and "which" on any other system. ` +
-      `If you are on Windows, please refer to: https://superuser.com/questions/49104/how-do-i-find-the-location-of-an-executable-in-window.\n`;
-  }
-
-  // debug(requirement, results);
-
-  for (let program of Object.keys(requirement)) {
+  for (let program of Object.keys(requirements)) {
+    const result = { path: which(program) };
     let message = '';
-
-    let req = requirement[program];
-    let res = results[program];
-
-    if (res?.path && (!req.version || semver.satisfies(res.version, req.version))) {
-      message += `✓  ${program}\n    Found at "${res.path}"` + (req.version ? ` (v${res.version} satisfies ${req.version})` : ``);
-      res.success = true;
-      if (res.multiple) {
-        message += `\n    Warning: multiple path found while checking.`;
+    let requirement = requirements[program];
+
+    if (result.path) {
+      if (requirement.version) {
+        result.version = await getVersion(program);
+        if (semver.satisfies(result.version, requirement.version)) {
+          message += `✓  ${program}\n    Found at "${result.path}" (v${result.version} satisfies ${requirement.version})`;
+          result.success = true;
+        }
+        else {
+          message += `x  "${program}"\n Installed but the version v${result.version} does not satisfies the requirement "${requirement.version}". ` +
+            `Please upgrade to the required version.`;
+          result.success = false;
+        }
+      }
+      else {
+        message += `✓  ${program}\n    Found at "${result.path}"`;
+        result.success = true;
       }
-    } else if (res?.path) {
-      message += `¯\\_(ツ)_/¯ "${program}" installed but old. Version is ${res.version} but we recommend ${req.version}. ` +
-        `Your version might or might not work. We strongly recommend upgrading to latest (or at least a later) version instead.`;
-      // success = false;
-    } else if (res) {
-      message += `x    ${program} not found.`;
-      res.success = success = false;
-    } else {
-      message += `?    ${program} not tested.`;
+    }
+    else {
+      message += `x  ${program}\n    Not found.`;
+      result.success = false;
+    }
+
+    if (!result.success) {
       success = false;
-      requirement[program] = { success: false };
     }
 
     modalMessage += `${message}\n`;
   }
 
   modalMessage += success ?
-    `\nSUCCESS! All system dependencies seem to be in order.` :
-    `\nPROBLEM: One or more system dependencies are not installed. Fix them, then try again.`;
-
-  // debug(success, modalMessage);
+    `\nSUCCESS! All system dependencies have been installed.` :
+    `\nFAILED: One or more requirement does not satisfied.`;
```

Comment:
`One or more requirement does not satisfied` -> `One or more requirements have not been satisfied.`